feeds/wlan-ap/opensync/src/platform/openwrt/src/netifd/src/wifi_inet_config.c

```diff
@@ -109,15 +110,13 @@ static void wifi_inet_conf_load(struct uci_section *s)
 
 	SCHEMA_SET_STR(conf.if_name, s->e.name);
 	if (tb[NET_ATTR_TYPE])
-                if ((strstr(s->e.name,"wan_") != NULL) || (strstr(s->e.name,"lan_") != NULL))
-                        SCHEMA_SET_STR(conf.if_type, "vlan");
-                else
-			SCHEMA_SET_STR(conf.if_type, blobmsg_get_string(tb[NET_ATTR_TYPE]));
+		SCHEMA_SET_STR(conf.if_type, blobmsg_get_string(tb[NET_ATTR_TYPE]));
```

Comment:
What case is this changing?

Reply:
On the way, I found this existing issue. In case of vlans on wan/lan, if_type inside Inet_Config was getting updated to eth instead of vlan because the condition "((strstr(s->e.name,"wan_") != NULL) || (strstr(s->e.name,"lan_") != NULL))" was kept inside "if (tb[NET_ATTR_TYPE])" and we do not set any "type" for vlan interfaces in /etc/config/network

Reply:
So what impact does changing this have?  Did you run it past Mike Hanson?

On my ECW5410, without your change, the Wifi_Inet_Config shows wan_100 and lan_100 as if_type vlan.  
However, on the Wifi_Inet_State I see wan_100 reported as vlan and lan_100 reported as eth.

Reply:
You get to see this issue when you restart opensync or reboot your AP, especially in a stand alone setup. When opensync starts, it would load the /etc/config/network and write to Wifi_Inet_Config table. Without the fix, if_type would be set to eth and not vlan because /etc/config/network does not have "type" set in case of vlans